homeassistant/components/switchbot/config_flow.py

```diff
@@ -0,0 +1,172 @@
+
+    if not switchbot_device:
+        raise NotConnectedError("Failed to discover switchbot")
+
+    return switchbot_device
+
+
+class SwitchbotConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Switchbot."""
+
+    VERSION = 1
+
+    async def _validate_mac(self, data) -> None:
+        """Try to connect to Switchbot device and create entry if successful."""
+        await self.async_set_unique_id(data[CONF_MAC].replace(":", ""))
+        self._abort_if_unique_id_configured()
+
+        # Validate bluetooth device mac.
+        # CONNECT_LOCK prevents btle adapter exceptions if there are multiple calls to this method.
+        try:
+            async with CONNECT_LOCK:
+                _btle_adv_data = await self.hass.async_add_executor_job(
+                    _btle_connect, data[CONF_MAC]
+                )
+
+        except NotConnectedError as err:
+            raise NotConnectedError(err) from err
+
+        if _btle_adv_data["modelName"] == "WoHand":
+            data[CONF_SENSOR_TYPE] = ATTR_BOT
+            return self.async_create_entry(title=data[CONF_NAME], data=data)
+
+        return self.async_abort(reason="switchbot_unsupported_type")
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry) -> None:
+        """Get the options flow for this handler."""
+        return SwitchbotOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None) -> None:
+        """Handle a flow initiated by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            user_input[CONF_MAC] = user_input[CONF_MAC].replace("-", ":").lower()
+
+            # abort if already configured.
+            for item in self._async_current_entries():
+                if item.data.get(CONF_MAC) == user_input[CONF_MAC]:
+                    return self.async_abort(reason="already_configured_device")
+
+            try:
+                return await self._validate_mac(user_input)
+
+            except NotConnectedError:
+                errors["base"] = "cannot_connect"
+
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                return self.async_abort(reason="unknown")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_NAME): str,
+                vol.Optional(CONF_PASSWORD): str,
+                vol.Required(CONF_MAC): str,
+            }
+        )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_import(self, import_config) -> None:
+        """Handle config import from yaml."""
+        _LOGGER.debug("import config: %s", import_config)
+
+        import_config[CONF_MAC] = import_config[CONF_MAC].replace("-", ":").lower()
+
+        await self.async_set_unique_id(import_config[CONF_MAC].replace(":", ""))
```

Comment:
Can you use `format_mac` here?
https://developers.home-assistant.io/docs/entity_registry_index/#example-acceptable-sources-for-a-unique-id

Reply:
format_mac won't work as I intentionally remove ":" in order to match with coordinator provided data.